Ecobee detect Smart Away

dcrypt3d opened the pull request:
**Description:**
Ecobee component is not able determine if the thermostat is in Away mode once Smart Away mode has created an away event. When Smart Away mode is enabled, after some time that occupancy is no longer detected, an event will be created with type = autoAway and holdClimateRef = ' '. When Away mode is manually enabled or set via Home-Assistant, type = hold and holdClimateRef = away.

This is a difficult situation to test but I was looking for some more validation and input on this code because I'm sure it's butchered, although it appears to work properly. There may be other cases I am currently not aware of. Normally my device trackers will Ecobee Away mode on and off. I had suspected this was happening months ago but was only able to confirm it while I was on vacation and someone was house sitting for me. 

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>
Somewhat related to #4510 
**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#<home-assistant.github.io PR number goes here>

**Example entry for `configuration.yaml` (if applicable):**
```yaml

```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51

dcrypt3d committed:
Ecobee autoAway Event

balloob commented on homeassistant/components/climate/ecobee.py:
```diff
@@ -195,9 +195,12 @@ def is_away_mode_on(self):
         events = self.thermostat['events']
         for event in events:
-            if event['running']:
+            if event['running'] and event['holdClimateRef'] == 'away':
```
Since both if statements result in `mode` being set to `away`, please merge both into 1 statement.

dcrypt3d committed:
Update ecobee.py

Checking if event['running'] true is pointless because if false event['type'] will equal template and when true type will only be 'hold' or 'autoAway' so I've removed this check from the statement

balloob commented:
Top 🐬

Duoxilian commented:
I think the issue is that there is an inconsistency in the implementation of is_away_mode_on, see issue #4953.

We should clarify what away_mode refers to. In my interpretation, it refers to that the thermostat is in a hold or smart away mode. The schedule itself is detected with self.mode. These all work fine, except we should remove the check for self.mode from the implementation of is_away_mode_on.

Duoxilian commented:
As an aside, I implemented the corresponding smarthome mode in issue #4950 as a trivial extension. It works and is handy if you want to manipulate that mode, e.g., putting the thermostat into home hold when the mode is sleep (to increase the temperature while you are up at that schedule).

devdelay commented:
is_away_mode_on refers to whether away is on or off set as a hold in events. To detect if Smart Away mode is enabled or change it, it is under the thermostat settings as "autoAway" true or false.

Functions to test and change Smart Away could probably be added pretty easily but I can't say I think it's really important for most users. I enabled Smart Away when I got the thermostat and I've not changed it since. I view Smart Away more as a catchall in the case that your automation did not trigger the thermostat into away, it will still go into away after a period of inactivity.

The purpose of this PR was to make Homeassistant display if away mode was on or off whether it was set manually or by timeout when autoAway is true. Previously it would only display if an away mode event had been set manually either at the thermostat or by Homeassistant but did not display that the thermostat was away if Smart Away set the thermostat using the autoAway event.
